### ISSUE TITLE
EditDistance.calculateMatrix fails with OutOfMemoryError for large arguments

### ISSUE THREAD

**Issue by leonard84** [spockframework/spock#121]
Since `EditDistance.calculateMatrix` allocates a two-dimensional int array which requires `seq1.length*seq2.length*4 byte` it can crash the JVM for large Strings. We should add a safety check and skip the distance calculation if the arguments are too large.

```
java.lang.OutOfMemoryError: Java heap space 

    at org.spockframework.runtime.condition.EditDistance.calculateMatrix(EditDistance.java:47)
    at org.spockframework.runtime.condition.EditDistance.<init>(EditDistance.java:42)
    at org.spockframework.runtime.ExpressionInfoValueRenderer.renderAsFailedStringComparison(ExpressionInfoValueRenderer.java:104)
    at org.spockframework.runtime.ExpressionInfoValueRenderer.doRenderValue(ExpressionInfoValueRenderer.java:88)
    at org.spockframework.runtime.ExpressionInfoValueRenderer.renderValue(ExpressionInfoValueRenderer.java:61)
    at org.spockframework.runtime.ExpressionInfoValueRenderer.render(ExpressionInfoValueRenderer.java:38)
    at org.spockframework.runtime.ExpressionInfoValueRenderer.render(ExpressionInfoValueRenderer.java:33)
    at org.spockframework.runtime.Condition.createRendering(Condition.java:94)
    at org.spockframework.runtime.Condition.getRendering(Condition.java:79)
    at org.spockframework.runtime.SpockComparisonFailure.getMessage(SpockComparisonFailure.java:41)
    at org.spockframework.runtime.SpockComparisonFailure.toString(SpockComparisonFailure.java:46)
    at java.lang.String.valueOf(String.java:2982)
    at java.io.PrintWriter.println(PrintWriter.java:754)
    at java.lang.Throwable$WrappedPrintWriter.println(Throwable.java:764)
    at java.lang.Throwable.printStackTrace(Throwable.java:655)
    at java.lang.Throwable.printStackTrace(Throwable.java:721)
    at com.intellij.rt.execution.junit.ComparisonDetailsExtractor.createPacket(ComparisonDetailsExtractor.java:97)
    at com.intellij.junit4.JUnit4TestResultsSender.doAddFailure(JUnit4TestResultsSender.java:88)
    at com.intellij.junit4.JUnit4TestResultsSender.testFailure(JUnit4TestResultsSender.java:57)
    at org.junit.runner.notification.SynchronizedRunListener.testFailure(SynchronizedRunListener.java:63)
    at org.junit.runner.notification.RunNotifier$4.notifyListener(RunNotifier.java:142)
    at org.junit.runner.notification.RunNotifier$SafeNotifier.run(RunNotifier.java:72)
    at org.junit.runner.notification.RunNotifier.fireTestFailures(RunNotifier.java:138)
    at org.junit.runner.notification.RunNotifier.fireTestFailure(RunNotifier.java:132)
    at org.spockframework.runtime.JUnitSupervisor.error(JUnitSupervisor.java:100)
    at org.spockframework.runtime.BaseSpecRunner.invokeRaw(BaseSpecRunner.java:482)
    at org.spockframework.runtime.BaseSpecRunner.invoke(BaseSpecRunner.java:463)
    at org.spockframework.runtime.BaseSpecRunner.runFeatureMethod(BaseSpecRunner.java:405)
    at org.spockframework.runtime.BaseSpecRunner.doRunIteration(BaseSpecRunner.java:324)
```

**Comment by askucins** [comment]
Hi, I'm getting the very same error in spock 1.1-groovy-2.4-rc-3 - does it mean that that fis has not been added to the spock codebase? If so - is there any hope for that?
Thanks a lot,
Adam

**Comment by askucins** [comment]
Thank you very much! That saved me a lot of troubles with my tests.

**Comment by timotei** [comment]
@leonard84 It seems that this is not fixed in latest spock: `1.1-groovy-2.4-rc-2`
Should I be using a different version? Or maybe it's not released yet?
